src/main/scala/cognite/spark/v1/MetricsBackend.scala

```diff
@@ -5,18 +5,24 @@ import cats.syntax.all._
 import com.codahale.metrics.Counter
 import sttp.capabilities.Effect
 import sttp.client3.{Request, Response, SttpBackend}
+import sttp.model.StatusCode
 import sttp.monad.MonadError
 
 import scala.language.higherKinds
 
 class MetricsBackend[F[_]: Sync, +S](
     delegate: SttpBackend[F, S],
-    metric: Counter
+    metric: Counter,
+    responseCounter: StatusCode => Option[Counter]
 ) extends SttpBackend[F, S] {
   override def send[T, R >: S with Effect[F]](
       request: Request[T, R]
   ): F[Response[T]] =
-    Sync[F].delay(metric.inc()) *> delegate.send(request)
+    Sync[F].delay(metric.inc()) *> delegate.send(request).flatTap { response =>
```

Comment:
That's a good point, I added a case for this. Right now we are mostly interested in 429 and 503s, but it may be useful for monitoring different kind of errors.